### ISSUE TITLE
Issue with Keycloak PostgreSQL Pod Permission Error on OpenShift using Microcks Operator

### ISSUE THREAD

**Issue by knighthl** [microcks/microcks#1269]
### Discussed in https://github.com/orgs/microcks/discussions/1266

<div type='discussions-op-text'>

<sup>Originally posted by **knighthl** August 14, 2024</sup>
Hi Microcks team,

I've encountered a permission issue when deploying Microcks using the Microcks Operator on OpenShift. Specifically, the error occurs in the my-microcksinstall-keycloak-postgresql pod during the initialization process.

Error Details:
```
chmod: /var/lib/postgresql/data: Operation not permitted
chmod: /var/run/postgresql: Operation not permitted
The files belonging to this database system will be owned by user "1000730000".
This user must also own the server process.
The database cluster will be initialized with locale "en_US.utf8".
The default database encoding has accordingly been set to "UTF8".
The default text search configuration will be set to "english".
Data page checksums are disabled.
fixing permissions on existing directory /var/lib/postgresql/data ... initdb: error: could not change permissions of directory "/var/lib/postgresql/data": Operation not permitted
```
CRD Configuration:

```
apiVersion: microcks.github.io/v1alpha1
kind: MicrocksInstall
metadata:
  name: my-microcksinstall
spec:
  name: my-microcksinstall
  version: 1.10.0
  microcks:
    replicas: 1
  postman:
    replicas: 1
  keycloak:
    install: true
    persistent: true
    volumeSize: 1Gi
  mongodb:
    install: true
    persistent: true
    volumeSize: 2Gi
```
**Environment**:
OpenShift Version: 4.14
Microcks Operator Version: 1.10.0

**Questions**:
Is this a known issue with the Microcks Operator on OpenShift, particularly with the Keycloak PostgreSQL pod?
Are there recommended configurations or workarounds to prevent this permission error from occurring?

Any guidance or suggestions would be greatly appreciated.

Thanks!</div>

**Comment by lbroudoux** [comment]
Hey there!

As i have no OpenShift cluster at hand, it's impossible for me to debug at the moment.
May I ask you to try something and report feedback here?

1) Launch the operator and the Kube objects initialized
2) Scale-down operator deployment to 0 to allow manual edits
3) On the `keycloak-postgresql` deployment, change the `volumeMount` directive to this one, adding a `subPath`:
```yaml
        volumeMounts:
          - name: "my-microcksinstall-keycloak-postgresql-data"
            mountPath: "/var/lib/postgresql/data"
            subPath: pgdata
```
4) Then still on the `keycloak-postgresql` deployment, add a new environment variable like this:
```yaml
        env:
            - name: PGDATA
              value: /var/lib/postgresql/data/pgdata
```

Let me know if it changes something.

**Comment by lbroudoux** [comment]
Hey there! I just installed an OpenShift Local instance (see https://docs.redhat.com/en/documentation/red_hat_openshift_local/2.40/html/getting_started_guide/index) and get the bug reproduced. Also, I tried the above fix and looks like it works well!
I'll try to push a new version of the operator during the week-end. I'll move the tag so you'll probably just have to redeploy it to get the update.

**Comment by lbroudoux** [comment]
For the record, I found useful information here: https://github.com/docker-library/postgres/issues/933 and use the solution that is suggested there: https://devops.stackexchange.com/questions/10669/cant-get-either-postgres-permissions-or-pvc-working-in-aks

**Comment by lbroudoux** [comment]
Issue will be actually tracked into microcks/microcks-ansible-operator#119. This repo will hold #1271 that is the Helm chart variant.

**Comment by lbroudoux** [comment]
Now shipped in operator `1.10.0-fix-1` tagged with `1.10.0`
